test/powershell/Language/Parser/Parser.Tests.ps1

```diff
@@ -915,4 +915,35 @@ foo``u{2195}abc
         # Issue #2780
         { ExecuteCommand "`$herestr=@`"`n'`"'`n`"@" } | Should Not Throw
     }
+
+    Context "#requires nested scan tokenizer tests" {
+        BeforeAll {
+            $settings = [System.Management.Automation.PSInvocationSettings]::new()
+            $settings.AddToHistory = $true
+
+            $ps = [powershell]::Create()
+        }
+
+        AfterEach {
+            $ps.Commands.Clear()
+        }
+
+        $testCases = @(
+            @{ script = "#requires"; firstToken = $null; lastToken = $null },
+            @{ script = "#requires -Version 5.0`n10"; firstToken = "10"; lastToken = "10" },
+            @{ script = "Write-Host 'Hello'`n#requires -Version 5.0`n7"; firstToken = "Write-Host"; lastToken = "7" },
+            @{ script = "Write-Host 'Hello'`n#requires -Version 5.0"; firstToken = "Write-Host"; lastToken = "Hello"}
+        )
+
+        It "Correctly resets the first and last tokens in the tokenizer after nested scan in script" -TestCases $testCases {
+            param($script, $firstToken, $lastToken)
+
+            $ps.AddScript($script)
+            $ps.AddScript("(`$^,`$`$)")
```

Comment:
I'm a little surprised this works, 2 calls to AddScript is I think like piping the first call to the second. That said, it neatly drops any results from the first call.

To be slightly more like the interactive scenario, I think you would call `$ps.Invoke` twice, clearing the commands between calls.

Reply:
Yeah, I tried that and it didn't seem to do what I wanted. To be honest, I wasn't entirely certain how I should be using the embedded PowerShell session to recreate the input. But I spent some time playing with it, tried the "script, invoke, tokens, invoke" way and seemed to get `$null` for everything.